File: lista 15questoes/15 questoes em python/att09.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def imprimir(self):
        print("Imprimindo Relatório")

class Contrato(Imprimivel):
    def imprimir(self):
        print("Imprimindo Contrato")

relatorio = Relatorio()
contrato = Contrato()
relatorio.imprimir()
contrato.imprimir()

File: lista 15questoes/15 questoes em java/att09.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo Relatório");
    }
}

class Contrato implements Imprimivel {
    public void imprimir() {
        System.out.println("Imprimindo Contrato");
    }
}

Relatorio relatorio = new Relatorio();
Contrato contrato = new Contrato();
relatorio.imprimir();
contrato.imprimir();
